src/systemctl/systemctl.c

```diff
@@ -9212,6 +9212,12 @@ static int systemctl_main(int argc, char *argv[]) {
                 {}
         };
 
+        const Verb *verb = verbs_find_verb(argv[optind], verbs);
+        if (verb && (verb->flags & VERB_ONLINE_ONLY) && arg_root)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                       "Verb '%s' cannot be used with --root=.",
+                                       argv[optind] ?: verb->verb);
+
```

Comment:
hmm, another option would be to simply set the `$SYSTEMD_OFFLINE` env var when `--root=` is specified. The right thing happens then anyway...

I think setting the env var is slightly nicer. dunno?

Reply:
The opposite for me ;) think the SYSTEMD_OFFLINE thing is very vague, because for most operations, it makes no sense to do nothing when offline. For example, `systemctl log-level` returns "", which is not one of the values in the enumeration and will most likely confuse any consumer. I think it'd be more appropriate to refuse operation. But handling compatiblity will existing users of this is hard, and I didn't want to open that can of worms. So I very much prefer to not use the variable internally if that's OK.

Reply:
the reason i preferred the $SYSTEMD_OFFLINE approach was that we now iterate through the verbs list twice.., once early, beforehand and once for real. hat's not sexy...

Maybe another option is to add a flags param to dispatch_verb forcing offline mode detection, which we could set here depending on arg_root. Then we'd iterate trough everything once only, and the errors about online/offline would be generated at a single place only.